libraries/Wire/src/Wire.h

```diff
@@ -56,8 +58,24 @@ class TwoWire : public Stream
     static void onRequestService(void);
     static void onReceiveService(uint8_t*, int);
 
-    uint8_t *allocateBuffer(uint8_t *buffer, size_t length);
-    uint8_t *resetBuffer(uint8_t *buffer);
+    inline void allocateRxBuffer(size_t length){ if(rxBufferAllocated < length) {
+        // By default we allocate BUFFER_LENGTH bytes. It is the min size of the buffer.
+        if(length < BUFFER_LENGTH) { length = BUFFER_LENGTH; }
+        rxBuffer = (uint8_t *)realloc(rxBuffer, length * sizeof(uint8_t));
```

Comment:
should't you check for a MAX SIZE somewhere to report an error rather than allocating all the available heap memory ?

Reply:
I think about it but I would not add more check.
If all heap is used then buffer will be at nullptr then an error will be raised.